src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,34 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
+    private static final int SLEEP_MILLIS = 1000;
+
     private final Executor executor;
+    private Map<String, UserData> cache;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
+        this.cache = new ConcurrentHashMap<>();
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() -> cache.computeIfAbsent(userId, id -> {
+            sleep();
+            return new UserData(userId, "Details for " + id);
+        }), executor);
```

Comment:
this function in compute if absent could also be extracted to some private method